test/step-function.test.ts

```diff
@@ -5348,3 +5583,139 @@ test("call Step Function from another Step Function", () => {
     },
   });
 });
+
+test("call Step Function from another Step Function with dynamic input", () => {
+  const { stack } = initStepFunctionApp();
+  const machine1 = new ExpressStepFunction<{ value: string }, string>(
+    stack,
+    "machine1",
+    () => {
+      return "hello";
+    }
+  );
+
+  const definition = new ExpressStepFunction<
+    { value1: string },
+    SyncExecutionResult<string>
+  >(stack, "machine2", (input) => {
+    const result = machine1({
+      input: {
+        value: input.value1,
+      },
+    });
+    return result;
+  }).definition;
+
+  expect(definition).toEqual({
+    StartAt: "result = machine1({input: {value: input.value1}})",
+    States: {
+      "result = machine1({input: {value: input.value1}})": {
+        Next: "return result",
+        Parameters: {
+          Input: {
+            "value.$": "$.value1",
+          },
+          StateMachineArn: machine1.stateMachineArn,
+        },
+        Resource: "arn:aws:states:::aws-sdk:sfn:startSyncExecution",
+        ResultPath: "$.result",
+        Type: "Task",
+      },
+      "return result": {
+        End: true,
+        OutputPath: "$.result",
+        Type: "Pass",
+      },
+    },
+  });
+});
+
+test("call Step Function from another Step Function not supported with reference argument", () => {
+  const { stack } = initStepFunctionApp();
+  const machine1 = new ExpressStepFunction<{ value: string }, string>(
+    stack,
+    "machine1",
+    () => {
+      return "hello";
+    }
+  );
+
+  expect(
```

Comment:
We should look into refactoring the Step Function test cases into a utility function that just takes the closure - that way we can migrate all of these test cases to use https://localstack.cloud one day.

Reply:
#97